onecodex/models/collection.py

```diff
@@ -274,6 +289,25 @@ def _collate_results(self, field=None, include_host=None):
             table = results["table"]
             host_tax_ids = results.get("host_tax_ids", [])
 
+            table = {t.get("tax_id"): t for t in table}
+
+            for tax_id, result in table.items():
+                if "abundance" not in result or result["abundance"] is None:
+                    result["abundance_w_children"] = 0
+                    continue
+
+                parent = table[result["parent_tax_id"]]
+                result["abundance_w_children"] = result["abundance"]
+
+                while parent:
+                    if "abundance_w_children" not in parent:
+                        parent["abundance_w_children"] = 0
+
+                    parent["abundance_w_children"] += result["abundance"]
+                    parent = table.get(parent["parent_tax_id"])
+
+            table = [v for k, v in table.items()]
```

Comment:
`table.values()`? 
Can we have a different name for the dict version since it doesn't have the same type?